// jsonresume-theme-even/dist/index.js
var y = Object.freeze, D = Object.defineProperty;
var b = (t, e) => y(D(t, "raw", { value: y(e || t.slice()) }));
import { html as i } from "@rbardini/html";
import L from "micromark";
import j from "striptags";
import k from "feather-icons";
const C = `const pluralize = (num, str) => \`\${num} \${num === 1 ? str : str.concat('s')}\`

class TimeDuration extends HTMLElement {
  connectedCallback() {
    const durationAttr = +this.getAttribute('duration')
    if (!durationAttr) return this.remove()

    const diffDate = new Date(durationAttr)
    const years = diffDate.getFullYear() - 1970
    const months = diffDate.getMonth()
    const days = diffDate.getDate() - 1

    const segments = [
      years && pluralize(years, 'yr'),
      months && pluralize(months, 'mo'),
      days && !years && !months && pluralize(days, 'day'),
    ].filter(Boolean)
    if (!segments.length) return

    this.textContent = segments.join(' ')
  }
}

customElements.define('time-duration', TimeDuration)
`, I = ':root{color-scheme:light dark;--color-background-light: #ffffff;--color-dimmed-light: #f3f4f5;--color-primary-light: #191e23;--color-secondary-light: #6c7781;--color-accent-light: #0073aa;--color-background-dark: #191e23;--color-dimmed-dark: #23282d;--color-primary-dark: #fbfbfc;--color-secondary-dark: #ccd0d4;--color-accent-dark: #00a0d2;--color-background: var(--color-background-light);--color-dimmed: var(--color-dimmed-light);--color-primary: var(--color-primary-light);--color-secondary: var(--color-secondary-light);--color-accent: var(--color-accent-light);--scale-ratio: 1.25;--scale0: 1rem;--scale1: calc(var(--scale0) * var(--scale-ratio));--scale2: calc(var(--scale1) * var(--scale-ratio));--scale3: calc(var(--scale2) * var(--scale-ratio));--scale4: calc(var(--scale3) * var(--scale-ratio));--scale5: calc(var(--scale4) * var(--scale-ratio))}@media (prefers-color-scheme: dark){:root{--color-background: var(--color-background-dark);--color-dimmed: var(--color-dimmed-dark);--color-primary: var(--color-primary-dark);--color-secondary: var(--color-secondary-dark);--color-accent: var(--color-accent-dark)}}*{box-sizing:border-box;margin:0;padding:0}html{font-size:14px}body{background:var(--color-background);color:var(--color-primary);display:grid;font:1em/1.5 Lato,sans-serif;gap:2em;grid-template-columns:[full-start] 1fr [main-start side-start] minmax(min-content,12em) [side-end content-start] minmax(min-content,36em) [main-end content-end] 1fr [full-end];grid-template-rows:auto [content] 0;margin-bottom:4em}body:before{content:"";grid-column:full;grid-row:content}ol,ul{padding-left:1em}:not(.icon-list,.tag-list)>li+li{margin-top:.4em}li::marker,.network{color:var(--color-secondary)}a{color:var(--color-accent);text-decoration:none}a:focus,a:hover{text-decoration:underline}h1,h2,h3,h5{font-weight:400}h1,h2,h3{line-height:1.2}h1{font-size:var(--scale5)}h2{color:var(--color-secondary);font-size:var(--scale4)}h3{color:var(--color-secondary);font-size:var(--scale3);grid-column:side;margin-bottom:1rem}h4{font-size:var(--scale2)}h5{font-size:var(--scale1)}h6{font-size:var(--scale0)}blockquote{border-left:.2em solid var(--color-dimmed);padding-left:1em}cite{color:var(--color-secondary);font-style:inherit}cite:before{content:"— "}svg{margin-right:.2em;vertical-align:text-bottom}.masthead{background:var(--color-dimmed);display:inherit;gap:inherit;grid-column:full;grid-template-columns:inherit;padding:4em 0;text-align:center}.masthead>*,section{grid-column:main}.masthead>img{border:4px solid;border-radius:50%;margin:0 auto;max-width:12em}article>*+*,blockquote>*+*,.timeline>div>*+*{margin-top:.6em}.meta{color:var(--color-secondary)}.stack{display:grid;gap:1.5em}.icon-list{display:flex;flex-wrap:wrap;gap:.4em 1em;justify-content:center;list-style:none;padding:0}.grid-list{display:grid;gap:1em}.tag-list{display:flex;flex-wrap:wrap;gap:.4em;list-style:none;padding:0}.tag-list>li{background:var(--color-dimmed);border-radius:.2em;padding:.2em .6em}.timeline>div{position:relative}.timeline>div:not(:last-child){padding-bottom:1rem}.timeline>div:not(:last-child):before{content:"";position:absolute;top:1rem;left:-15px;width:2px;height:100%;background:var(--color-secondary)}.timeline>div:not(:only-child):after{content:"";position:absolute;top:.6rem;left:-20px;width:8px;height:8px;background:var(--color-secondary);border:2px solid var(--color-background);border-radius:50%}@media print,(min-width: 48em){h3{text-align:right;margin-bottom:inherit}.masthead{text-align:inherit}.masthead>*,section{grid-column:content}.masthead img{grid-column:side;grid-row:span 2;max-width:100%}section{display:contents}.icon-list{flex-direction:column}.grid-list{grid-template-columns:repeat(auto-fit,minmax(calc((100% - 1em)/2),1fr))}}time+time-duration:before{content:"· "}@media print{time-duration{display:none}}';
function A(t = {}) {
  const e = t.themeOptions?.colors;
  return e && Object.entries(e).map(([o, [n, r = n]]) => `--color-${o}-light:${n}; --color-${o}-dark:${r};`).join(" ");
}
function d(t, e = !1) {
  const o = (
    /** @type {string} */
    L(t)
  );
  return e ? j(o) : o;
}
const T = (t) => new Date(t).toLocaleDateString("en", {
  month: "short",
  year: "numeric",
  timeZone: "UTC"
});
function u(t) {
  return i`<time datetime="${t}">${T(t)}</time>`;
}
function P(t = []) {
  return t.length > 0 && i`
      <section id="awards">
        <h3>Awards</h3>
        <div class="stack">
          ${t.map(
    ({ awarder: e, date: o, summary: n, title: r }) => i`
              <article>
                <header>
                  <h4>${r}</h4>
                  <div class="meta">
                    ${e && i`<div>Awarded by <strong>${e}</strong></div>`} ${o && u(o)}
                  </div>
                </header>
                ${n && d(n)}
              </article>
            `
  )}
        </div>
      </section>
    `;
}
const R = (t) => t.replace(/^(https?:|)\/\//, "").replace(/\/$/, "");
function h(t, e) {
  return e ? t ? i`<a href="${t}">${e}</a>` : e : t && i`<a href="${t}">${R(t)}</a>`;
}
function q(t = []) {
  return t.length > 0 && i`
      <section id="certificates">
        <h3>Certificates</h3>
        <div class="stack">
          ${t.map(
    ({ date: e, issuer: o, name: n, url: r }) => i`
              <article>
                <header>
                  <h4>${h(r, n)}</h4>
                  <div class="meta">
                    ${o && i`<div>Issued by <strong>${o}</strong></div>`} ${e && u(e)}
                  </div>
                </header>
              </article>
            `
  )}
        </div>
      </section>
    `;
}
function w(t) {
  const e = t.reduce(
    (o, { startDate: n, endDate: r }) => o + (n ? +new Date(r || Date.now()) - +new Date(n) : 0),
    0
  );
  return i`${e > 0 && i`<time-duration duration="${e}"></time-duration>`}`;
}
function p(t, e) {
  return e === t ? u(e) : i`${u(t)} – ${e ? u(e) : "Present"} ${w([{ startDate: t, endDate: e }])}`;
}
function B(t = []) {
  return t.length > 0 && i`
      <section id="education">
        <h3>Education</h3>
        <div class="stack">
          ${t.map(
    ({ area: e, courses: o = [], institution: n, startDate: r, endDate: l, studyType: a, url: c }) => i`
              <article>
                <header>
                  <h4>${h(c, n)}</h4>
                  <div class="meta">
                    <div>${[a, e && i`<strong>${e}</strong>`].filter(Boolean).join(" in ")}</div>
                    ${r && i`<div>${p(r, l)}</div>`}
                  </div>
                </header>
                ${o.length > 0 && i`
                  <h5>Courses</h5>
                  <ul>
                    ${o.map((s) => i`<li>${d(s)}</li>`)}
                  </ul>
                `}
              </article>
            `
  )}
        </div>
      </section>
    `;
}
function v(t, e) {
  return (k.icons[
    /** @type {FeatherIconNames} */
    t.toLowerCase()
  ] || e && k.icons[
    /** @type {FeatherIconNames} */
    e.toLowerCase()
  ])?.toSvg({ width: 16, height: 16 });
}
const E = (t) => Intl.DisplayNames ? new Intl.DisplayNames(["en"], { type: "region" }).of(t) : t;
function S(t = {}) {
  const { email: e, image: o, label: n, location: r, name: l, phone: a, profiles: c = [], summary: s, url: $ } = t;
  return i`
    <header class="masthead">
      ${o && i`<img src="${o}" alt="" />`}
      <div>${l && i`<h1>${l}</h1>`} ${n && i`<h2>${n}</h2>`}</div>
      ${s && i`<article>${d(s)}</article>`}
      <ul class="icon-list">
        ${r?.city && i`
          <li>
            ${v("map-pin")} ${r.city}${r.countryCode && i`, ${E(r.countryCode)}`}
          </li>
        `}
        ${e && i`
          <li>
            ${v("mail")}
            <a href="mailto:${e}">${e}</a>
          </li>
        `}
        ${a && i`
          <li>
            ${v("phone")}
            <a href="tel:${a.replace(/\s/g, "")}">${a}</a>
          </li>
        `}
        ${$ && i`<li>${v("link")} ${h($)}</li>`}
        ${c.map(
    ({ network: m, url: g, username: f }) => i`
            <li>
              ${m && v(m, "user")} ${h(g, f)}
              ${m && i`<span class="network">(${m})</span>`}
            </li>
          `
  )}
      </ul>
    </header>
  `;
}
function F(t = []) {
  return t.length > 0 && i`
      <section id="interests">
        <h3>Interests</h3>
        <div class="grid-list">
          ${t.map(
    ({ keywords: e = [], name: o }) => i`
              <div>
                ${o && i`<h4>${o}</h4>`}
                ${e.length > 0 && i`
                  <ul class="tag-list">
                    ${e.map((n) => i`<li>${n}</li>`)}
                  </ul>
                `}
              </div>
            `
  )}
        </div>
      </section>
    `;
}
function M(t = []) {
  return t.length > 0 && i`
      <section id="languages">
        <h3>Languages</h3>
        <div class="grid-list">
          ${t.map(
    ({ fluency: e, language: o }) => i`<div>${o && i`<h4>${o}</h4>`} ${e}</div>`
  )}
        </div>
      </section>
    `;
}
function O(t = {}) {
  const { name: e, summary: o } = t;
  return i`
    ${e && i`<title>${e}</title>`}
    ${o && i`<meta name="description" content="${d(o, !0)}" />`}
  `;
}
const W = (t) => Intl.ListFormat ? new Intl.ListFormat("en").format(t) : t.join(", ");
function H(t = []) {
  return t.length > 0 && i`
      <section id="projects">
        <h3>Projects</h3>
        <div class="stack">
          ${t.map(
    ({
      description: e,
      entity: o,
      highlights: n = [],
      keywords: r = [],
      name: l,
      startDate: a,
      endDate: c,
      roles: s = [],
      type: $,
      url: m
    }) => i`
              <article>
                <header>
                  <h4>${h(m, l)}</h4>
                  <div class="meta">
                    <div>
                      ${s.length > 0 && i`<strong>${W(s)}</strong>`}
                      ${o && i`at <strong>${o}</strong>`}
                    </div>
                    ${a && i`<div>${p(a, c)}</div>`}
                    ${$ && i`<div>${$}</div>`}
                  </div>
                </header>
                ${e && d(e)}
                ${n.length > 0 && i`
                  <ul>
                    ${n.map((g) => i`<li>${d(g)}</li>`)}
                  </ul>
                `}
                ${r.length > 0 && i`
                  <ul class="tag-list">
                    ${r.map((g) => i`<li>${g}</li>`)}
                  </ul>
                `}
              </article>
            `
  )}
        </div>
      </section>
    `;
}
function N(t = []) {
  return t.length > 0 && i`
      <section id="publications">
        <h3>Publications</h3>
        <div class="stack">
          ${t.map(
    ({ name: e, publisher: o, releaseDate: n, summary: r, url: l }) => i`
              <article>
                <header>
                  <h4>${h(l, e)}</h4>
                  <div class="meta">
                    ${o && i`<div>Published by <strong>${o}</strong></div>`}
                    ${n && u(n)}
                  </div>
                </header>
                ${r && d(r)}
              </article>
            `
  )}
        </div>
      </section>
    `;
}
function U(t = []) {
  return t.length > 0 && i`
      <section id="references">
        <h3>References</h3>
        <div class="stack">
          ${t.map(
    ({ name: e, reference: o }) => i`
              <blockquote>
                ${o && d(o)}
                ${e && i`
                  <p>
                    <cite>${e}</cite>
                  </p>
                `}
              </blockquote>
            `
  )}
        </div>
      </section>
    `;
}
function V(t = []) {
  return t.length > 0 && i`
      <section id="skills">
        <h3>Skills</h3>
        <div class="grid-list">
          ${t.map(
    ({ keywords: e = [], name: o }) => i`
              <div>
                ${o && i`<h4>${o}</h4>`}
                ${e.length > 0 && i`
                  <ul class="tag-list">
                    ${e.map((n) => i`<li>${n}</li>`)}
                  </ul>
                `}
              </div>
            `
  )}
        </div>
      </section>
    `;
}
function Y(t = []) {
  return t.length > 0 && i`
      <section id="volunteer">
        <h3>Volunteer</h3>
        <div class="stack">
          ${t.map(
    ({ highlights: e = [], organization: o, position: n, startDate: r, endDate: l, summary: a, url: c }) => i`
              <article>
                <header>
                  <h4>${h(c, o)}</h4>
                  <div class="meta">
                    <strong>${n}</strong>
                    ${r && i`<div>${p(r, l)}</div>`}
                  </div>
                </header>
                ${a && d(a)}
                ${e.length > 0 && i`
                  <ul>
                    ${e.map((s) => i`<li>${d(s)}</li>`)}
                  </ul>
                `}
              </article>
            `
  )}
        </div>
      </section>
    `;
}
function Z(t = []) {
  const e = t.reduce(
    (o, { description: n, name: r, url: l, ...a }) => {
      const c = o[o.length - 1];
      return c && c.name === r && c.description === n && c.url === l ? c.items.push(a) : o.push({ description: n, name: r, url: l, items: [a] }), o;
    },
    /** @type {NestedWork[]} */
    []
  );
  return t.length > 0 && i`
      <section id="work">
        <h3>Work</h3>
        <div class="stack">
          ${e.map(({ description: o, name: n, url: r, items: l = [] }) => {
    const a = l.length === 1 ? l[0] : void 0;
    return i`
              <article>
                <header>
                  <h4>${a ? a.position : h(r, n)}</h4>
                  <div class="meta">
                    ${a ? i`
                          <div>
                            ${[i`<strong>${h(r, n)}</strong>`, o].filter(Boolean).join(" · ")}
                          </div>
                          ${a.startDate && i`<div>${p(a.startDate, a.endDate)}</div>`}
                          ${a.location && i`<div>${a.location}</div>`}
                        ` : i`
                          ${o && i`<div>${o}</div>`}
                          ${l.some((c) => c.startDate) && i`<div>${w(l)}</div>`}
                        `}
                  </div>
                </header>
                <div class="timeline">
                  ${l.map(
      ({ highlights: c = [], location: s, position: $, startDate: m, endDate: g, summary: f }) => i`
                      <div>
                        ${!a && i`
                          <div>
                            <h5>${$}</h5>
                            <div class="meta">
                              ${m && i`<div>${p(m, g)}</div>`}
                              ${s && i`<div>${s}</div>`}
                            </div>
                          </div>
                        `}
                        ${f && d(f)}
                        ${c.length > 0 && i`
                          <ul>
                            ${c.map((z) => i`<li>${d(z)}</li>`)}
                          </ul>
                        `}
                      </div>
                    `
    )}
                </div>
              </article>
            `;
  })}
        </div>
      </section>
    `;
}
var x;
function G(t, { css: e, js: o } = {}) {
  return i`<!doctype html>
    <html lang="en" style="${A(t.meta)}">
      <head>
        <meta charset="utf-8" />
        ${O(t.basics)}
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700&display=swap" />
        ${e && i`<style>
          ${e}
        </style>`}
        ${o && i(x || (x = b([`<script type="module">
          `, `
        <\/script>`])), o)}
      </head>
      <body>
        ${S(t.basics)} ${Z(t.work)} ${Y(t.volunteer)} ${B(t.education)}
        ${H(t.projects)} ${P(t.awards)} ${q(t.certificates)}
        ${N(t.publications)} ${V(t.skills)} ${M(t.languages)}
        ${F(t.interests)} ${U(t.references)}
      </body>
    </html>`;
}
const tt = {
  mediaType: "print",
  printBackground: !0
}, it = (t) => G(t, { css: I, js: C });
export {
  tt as pdfRenderOptions,
  it as render
};
