// jsonresume-theme-even/dist/index.cjs
"use strict";var y=Object.freeze,D=Object.defineProperty;var b=(t,i)=>y(D(t,"raw",{value:y(i||t.slice())}));Object.defineProperty(exports,Symbol.toStringTag,{value:"Module"});const e=require("@rbardini/html"),L=require("micromark"),j=require("striptags"),k=require("feather-icons"),C=`const pluralize = (num, str) => \`\${num} \${num === 1 ? str : str.concat('s')}\`

class TimeDuration extends HTMLElement {
  connectedCallback() {
    const durationAttr = +this.getAttribute('duration')
    if (!durationAttr) return this.remove()

    const diffDate = new Date(durationAttr)
    const years = diffDate.getFullYear() - 1970
    const months = diffDate.getMonth()
    const days = diffDate.getDate() - 1

    const segments = [
      years && pluralize(years, 'yr'),
      months && pluralize(months, 'mo'),
      days && !years && !months && pluralize(days, 'day'),
    ].filter(Boolean)
    if (!segments.length) return

    this.textContent = segments.join(' ')
  }
}

customElements.define('time-duration', TimeDuration)
`,I=':root{color-scheme:light dark;--color-background-light: #ffffff;--color-dimmed-light: #f3f4f5;--color-primary-light: #191e23;--color-secondary-light: #6c7781;--color-accent-light: #0073aa;--color-background-dark: #191e23;--color-dimmed-dark: #23282d;--color-primary-dark: #fbfbfc;--color-secondary-dark: #ccd0d4;--color-accent-dark: #00a0d2;--color-background: var(--color-background-light);--color-dimmed: var(--color-dimmed-light);--color-primary: var(--color-primary-light);--color-secondary: var(--color-secondary-light);--color-accent: var(--color-accent-light);--scale-ratio: 1.25;--scale0: 1rem;--scale1: calc(var(--scale0) * var(--scale-ratio));--scale2: calc(var(--scale1) * var(--scale-ratio));--scale3: calc(var(--scale2) * var(--scale-ratio));--scale4: calc(var(--scale3) * var(--scale-ratio));--scale5: calc(var(--scale4) * var(--scale-ratio))}@media (prefers-color-scheme: dark){:root{--color-background: var(--color-background-dark);--color-dimmed: var(--color-dimmed-dark);--color-primary: var(--color-primary-dark);--color-secondary: var(--color-secondary-dark);--color-accent: var(--color-accent-dark)}}*{box-sizing:border-box;margin:0;padding:0}html{font-size:14px}body{background:var(--color-background);color:var(--color-primary);display:grid;font:1em/1.5 Lato,sans-serif;gap:2em;grid-template-columns:[full-start] 1fr [main-start side-start] minmax(min-content,12em) [side-end content-start] minmax(min-content,36em) [main-end content-end] 1fr [full-end];grid-template-rows:auto [content] 0;margin-bottom:4em}body:before{content:"";grid-column:full;grid-row:content}ol,ul{padding-left:1em}:not(.icon-list,.tag-list)>li+li{margin-top:.4em}li::marker,.network{color:var(--color-secondary)}a{color:var(--color-accent);text-decoration:none}a:focus,a:hover{text-decoration:underline}h1,h2,h3,h5{font-weight:400}h1,h2,h3{line-height:1.2}h1{font-size:var(--scale5)}h2{color:var(--color-secondary);font-size:var(--scale4)}h3{color:var(--color-secondary);font-size:var(--scale3);grid-column:side;margin-bottom:1rem}h4{font-size:var(--scale2)}h5{font-size:var(--scale1)}h6{font-size:var(--scale0)}blockquote{border-left:.2em solid var(--color-dimmed);padding-left:1em}cite{color:var(--color-secondary);font-style:inherit}cite:before{content:"— "}svg{margin-right:.2em;vertical-align:text-bottom}.masthead{background:var(--color-dimmed);display:inherit;gap:inherit;grid-column:full;grid-template-columns:inherit;padding:4em 0;text-align:center}.masthead>*,section{grid-column:main}.masthead>img{border:4px solid;border-radius:50%;margin:0 auto;max-width:12em}article>*+*,blockquote>*+*,.timeline>div>*+*{margin-top:.6em}.meta{color:var(--color-secondary)}.stack{display:grid;gap:1.5em}.icon-list{display:flex;flex-wrap:wrap;gap:.4em 1em;justify-content:center;list-style:none;padding:0}.grid-list{display:grid;gap:1em}.tag-list{display:flex;flex-wrap:wrap;gap:.4em;list-style:none;padding:0}.tag-list>li{background:var(--color-dimmed);border-radius:.2em;padding:.2em .6em}.timeline>div{position:relative}.timeline>div:not(:last-child){padding-bottom:1rem}.timeline>div:not(:last-child):before{content:"";position:absolute;top:1rem;left:-15px;width:2px;height:100%;background:var(--color-secondary)}.timeline>div:not(:only-child):after{content:"";position:absolute;top:.6rem;left:-20px;width:8px;height:8px;background:var(--color-secondary);border:2px solid var(--color-background);border-radius:50%}@media print,(min-width: 48em){h3{text-align:right;margin-bottom:inherit}.masthead{text-align:inherit}.masthead>*,section{grid-column:content}.masthead img{grid-column:side;grid-row:span 2;max-width:100%}section{display:contents}.icon-list{flex-direction:column}.grid-list{grid-template-columns:repeat(auto-fit,minmax(calc((100% - 1em)/2),1fr))}}time+time-duration:before{content:"· "}@media print{time-duration{display:none}}';function q(t={}){const i=t.themeOptions?.colors;return i&&Object.entries(i).map(([o,[n,l=n]])=>`--color-${o}-light:${n}; --color-${o}-dark:${l};`).join(" ")}function d(t,i=!1){const o=L(t);return i?j(o):o}const T=t=>new Date(t).toLocaleDateString("en",{month:"short",year:"numeric",timeZone:"UTC"});function g(t){return e.html`<time datetime="${t}">${T(t)}</time>`}function A(t=[]){return t.length>0&&e.html`
      <section id="awards">
        <h3>Awards</h3>
        <div class="stack">
          ${t.map(({awarder:i,date:o,summary:n,title:l})=>e.html`
              <article>
                <header>
                  <h4>${l}</h4>
                  <div class="meta">
                    ${i&&e.html`<div>Awarded by <strong>${i}</strong></div>`} ${o&&g(o)}
                  </div>
                </header>
                ${n&&d(n)}
              </article>
            `)}
        </div>
      </section>
    `}const P=t=>t.replace(/^(https?:|)\/\//,"").replace(/\/$/,"");function h(t,i){return i?t?e.html`<a href="${t}">${i}</a>`:i:t&&e.html`<a href="${t}">${P(t)}</a>`}function R(t=[]){return t.length>0&&e.html`
      <section id="certificates">
        <h3>Certificates</h3>
        <div class="stack">
          ${t.map(({date:i,issuer:o,name:n,url:l})=>e.html`
              <article>
                <header>
                  <h4>${h(l,n)}</h4>
                  <div class="meta">
                    ${o&&e.html`<div>Issued by <strong>${o}</strong></div>`} ${i&&g(i)}
                  </div>
                </header>
              </article>
            `)}
        </div>
      </section>
    `}function x(t){const i=t.reduce((o,{startDate:n,endDate:l})=>o+(n?+new Date(l||Date.now())-+new Date(n):0),0);return e.html`${i>0&&e.html`<time-duration duration="${i}"></time-duration>`}`}function p(t,i){return i===t?g(i):e.html`${g(t)} – ${i?g(i):"Present"} ${x([{startDate:t,endDate:i}])}`}function S(t=[]){return t.length>0&&e.html`
      <section id="education">
        <h3>Education</h3>
        <div class="stack">
          ${t.map(({area:i,courses:o=[],institution:n,startDate:l,endDate:a,studyType:r,url:c})=>e.html`
              <article>
                <header>
                  <h4>${h(c,n)}</h4>
                  <div class="meta">
                    <div>${[r,i&&e.html`<strong>${i}</strong>`].filter(Boolean).join(" in ")}</div>
                    ${l&&e.html`<div>${p(l,a)}</div>`}
                  </div>
                </header>
                ${o.length>0&&e.html`
                  <h5>Courses</h5>
                  <ul>
                    ${o.map(s=>e.html`<li>${d(s)}</li>`)}
                  </ul>
                `}
              </article>
            `)}
        </div>
      </section>
    `}function v(t,i){return(k.icons[t.toLowerCase()]||i&&k.icons[i.toLowerCase()])?.toSvg({width:16,height:16})}const O=t=>Intl.DisplayNames?new Intl.DisplayNames(["en"],{type:"region"}).of(t):t;function B(t={}){const{email:i,image:o,label:n,location:l,name:a,phone:r,profiles:c=[],summary:s,url:$}=t;return e.html`
    <header class="masthead">
      ${o&&e.html`<img src="${o}" alt="" />`}
      <div>${a&&e.html`<h1>${a}</h1>`} ${n&&e.html`<h2>${n}</h2>`}</div>
      ${s&&e.html`<article>${d(s)}</article>`}
      <ul class="icon-list">
        ${l?.city&&e.html`
          <li>
            ${v("map-pin")} ${l.city}${l.countryCode&&e.html`, ${O(l.countryCode)}`}
          </li>
        `}
        ${i&&e.html`
          <li>
            ${v("mail")}
            <a href="mailto:${i}">${i}</a>
          </li>
        `}
        ${r&&e.html`
          <li>
            ${v("phone")}
            <a href="tel:${r.replace(/\s/g,"")}">${r}</a>
          </li>
        `}
        ${$&&e.html`<li>${v("link")} ${h($)}</li>`}
        ${c.map(({network:m,url:u,username:f})=>e.html`
            <li>
              ${m&&v(m,"user")} ${h(u,f)}
              ${m&&e.html`<span class="network">(${m})</span>`}
            </li>
          `)}
      </ul>
    </header>
  `}function E(t=[]){return t.length>0&&e.html`
      <section id="interests">
        <h3>Interests</h3>
        <div class="grid-list">
          ${t.map(({keywords:i=[],name:o})=>e.html`
              <div>
                ${o&&e.html`<h4>${o}</h4>`}
                ${i.length>0&&e.html`
                  <ul class="tag-list">
                    ${i.map(n=>e.html`<li>${n}</li>`)}
                  </ul>
                `}
              </div>
            `)}
        </div>
      </section>
    `}function M(t=[]){return t.length>0&&e.html`
      <section id="languages">
        <h3>Languages</h3>
        <div class="grid-list">
          ${t.map(({fluency:i,language:o})=>e.html`<div>${o&&e.html`<h4>${o}</h4>`} ${i}</div>`)}
        </div>
      </section>
    `}function F(t={}){const{name:i,summary:o}=t;return e.html`
    ${i&&e.html`<title>${i}</title>`}
    ${o&&e.html`<meta name="description" content="${d(o,!0)}" />`}
  `}const W=t=>Intl.ListFormat?new Intl.ListFormat("en").format(t):t.join(", ");function H(t=[]){return t.length>0&&e.html`
      <section id="projects">
        <h3>Projects</h3>
        <div class="stack">
          ${t.map(({description:i,entity:o,highlights:n=[],keywords:l=[],name:a,startDate:r,endDate:c,roles:s=[],type:$,url:m})=>e.html`
              <article>
                <header>
                  <h4>${h(m,a)}</h4>
                  <div class="meta">
                    <div>
                      ${s.length>0&&e.html`<strong>${W(s)}</strong>`}
                      ${o&&e.html`at <strong>${o}</strong>`}
                    </div>
                    ${r&&e.html`<div>${p(r,c)}</div>`}
                    ${$&&e.html`<div>${$}</div>`}
                  </div>
                </header>
                ${i&&d(i)}
                ${n.length>0&&e.html`
                  <ul>
                    ${n.map(u=>e.html`<li>${d(u)}</li>`)}
                  </ul>
                `}
                ${l.length>0&&e.html`
                  <ul class="tag-list">
                    ${l.map(u=>e.html`<li>${u}</li>`)}
                  </ul>
                `}
              </article>
            `)}
        </div>
      </section>
    `}function N(t=[]){return t.length>0&&e.html`
      <section id="publications">
        <h3>Publications</h3>
        <div class="stack">
          ${t.map(({name:i,publisher:o,releaseDate:n,summary:l,url:a})=>e.html`
              <article>
                <header>
                  <h4>${h(a,i)}</h4>
                  <div class="meta">
                    ${o&&e.html`<div>Published by <strong>${o}</strong></div>`}
                    ${n&&g(n)}
                  </div>
                </header>
                ${l&&d(l)}
              </article>
            `)}
        </div>
      </section>
    `}function U(t=[]){return t.length>0&&e.html`
      <section id="references">
        <h3>References</h3>
        <div class="stack">
          ${t.map(({name:i,reference:o})=>e.html`
              <blockquote>
                ${o&&d(o)}
                ${i&&e.html`
                  <p>
                    <cite>${i}</cite>
                  </p>
                `}
              </blockquote>
            `)}
        </div>
      </section>
    `}function V(t=[]){return t.length>0&&e.html`
      <section id="skills">
        <h3>Skills</h3>
        <div class="grid-list">
          ${t.map(({keywords:i=[],name:o})=>e.html`
              <div>
                ${o&&e.html`<h4>${o}</h4>`}
                ${i.length>0&&e.html`
                  <ul class="tag-list">
                    ${i.map(n=>e.html`<li>${n}</li>`)}
                  </ul>
                `}
              </div>
            `)}
        </div>
      </section>
    `}function Y(t=[]){return t.length>0&&e.html`
      <section id="volunteer">
        <h3>Volunteer</h3>
        <div class="stack">
          ${t.map(({highlights:i=[],organization:o,position:n,startDate:l,endDate:a,summary:r,url:c})=>e.html`
              <article>
                <header>
                  <h4>${h(c,o)}</h4>
                  <div class="meta">
                    <strong>${n}</strong>
                    ${l&&e.html`<div>${p(l,a)}</div>`}
                  </div>
                </header>
                ${r&&d(r)}
                ${i.length>0&&e.html`
                  <ul>
                    ${i.map(s=>e.html`<li>${d(s)}</li>`)}
                  </ul>
                `}
              </article>
            `)}
        </div>
      </section>
    `}function Z(t=[]){const i=t.reduce((o,{description:n,name:l,url:a,...r})=>{const c=o[o.length-1];return c&&c.name===l&&c.description===n&&c.url===a?c.items.push(r):o.push({description:n,name:l,url:a,items:[r]}),o},[]);return t.length>0&&e.html`
      <section id="work">
        <h3>Work</h3>
        <div class="stack">
          ${i.map(({description:o,name:n,url:l,items:a=[]})=>{const r=a.length===1?a[0]:void 0;return e.html`
              <article>
                <header>
                  <h4>${r?r.position:h(l,n)}</h4>
                  <div class="meta">
                    ${r?e.html`
                          <div>
                            ${[e.html`<strong>${h(l,n)}</strong>`,o].filter(Boolean).join(" · ")}
                          </div>
                          ${r.startDate&&e.html`<div>${p(r.startDate,r.endDate)}</div>`}
                          ${r.location&&e.html`<div>${r.location}</div>`}
                        `:e.html`
                          ${o&&e.html`<div>${o}</div>`}
                          ${a.some(c=>c.startDate)&&e.html`<div>${x(a)}</div>`}
                        `}
                  </div>
                </header>
                <div class="timeline">
                  ${a.map(({highlights:c=[],location:s,position:$,startDate:m,endDate:u,summary:f})=>e.html`
                      <div>
                        ${!r&&e.html`
                          <div>
                            <h5>${$}</h5>
                            <div class="meta">
                              ${m&&e.html`<div>${p(m,u)}</div>`}
                              ${s&&e.html`<div>${s}</div>`}
                            </div>
                          </div>
                        `}
                        ${f&&d(f)}
                        ${c.length>0&&e.html`
                          <ul>
                            ${c.map(z=>e.html`<li>${d(z)}</li>`)}
                          </ul>
                        `}
                      </div>
                    `)}
                </div>
              </article>
            `})}
        </div>
      </section>
    `}var w;function G(t,{css:i,js:o}={}){return e.html`<!doctype html>
    <html lang="en" style="${q(t.meta)}">
      <head>
        <meta charset="utf-8" />
        ${F(t.basics)}
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700&display=swap" />
        ${i&&e.html`<style>
          ${i}
        </style>`}
        ${o&&e.html(w||(w=b([`<script type="module">
          `,`
        <\/script>`])),o)}
      </head>
      <body>
        ${B(t.basics)} ${Z(t.work)} ${Y(t.volunteer)} ${S(t.education)}
        ${H(t.projects)} ${A(t.awards)} ${R(t.certificates)}
        ${N(t.publications)} ${V(t.skills)} ${M(t.languages)}
        ${E(t.interests)} ${U(t.references)}
      </body>
    </html>`}const J={mediaType:"print",printBackground:!0},K=t=>G(t,{css:I,js:C});exports.pdfRenderOptions=J;exports.render=K;
